boswatch/network/server.py

```diff
@@ -59,15 +59,22 @@ def handle(self):
                 logging.debug("%s recv header: '%s'", req_name, header)
                 logging.debug("%s recv %d bytes:\n%s", req_name, len(data), data)
 
-                # add a new entry and the decoded data dict as an string in utf-8 and an timestamp
-                self.server.alarmQueue.put_nowait((self.client_address[0], data, time.time()))  # queue is threadsafe
-                logging.debug("Add data to queue")
-
-                logging.debug("%s send: [ack]", req_name)
-
-                data = "[ack]".encode("utf-8")
-                header = str(len(data)).ljust(HEADERSIZE).encode("utf-8")
-                self.request.sendall(header + data)
+                try:
+                    eval(str(data.strip()))
```

Comment:
Because the error mentioned in the issue occurs by executing this line of code later in packet.py, if a package is not transmitted completely:

Traceback (most recent call last): File "bw_server.py", line 101, in <module> bwPacket = Packet((data[1])) File "/opt/boswatch3/boswatch/packet.py", line 35, in init self._packet = **eval(str(bwPacket.strip()))** File "<string>", line 1 <keep-aliv ^ SyntaxError: invalid syntax

But i have to know, if i can execute this line, already after receiving the data, to decide, whether i have to send ack or nack.


If you have a better solution for checking, if the package was transmitted completely or not, feel free to implement it instead running the eval(str(data))

Reply:
If I see it correctly, this line is responsible for the automatic checks failing. That means we need to make some improvements here.
The current code line throws reliably in case of error the SyntaxError, but can we check this also differently, because without eval() the error is not thrown and therefore not handled!